pkg/controller/disruption/disruption.go

```diff
@@ -0,0 +1,486 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package disruption
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/apis/policy"
+	"k8s.io/kubernetes/pkg/client/cache"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/controller"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/types"
+	"k8s.io/kubernetes/pkg/util/intstr"
+	"k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/util/workqueue"
+	"k8s.io/kubernetes/pkg/watch"
+
+	"github.com/golang/glog"
+)
+
+type updater func(*policy.PodDisruptionBudget) error
+
+type DisruptionController struct {
+	kubeClient *client.Client
+
+	pdbStore      cache.Store
+	pdbController *framework.Controller
+	pdbLister     cache.StoreToPodDisruptionBudgetLister
+
+	podController framework.ControllerInterface
+	podLister     cache.StoreToPodLister
+
+	rcIndexer    cache.Indexer
+	rcController *framework.Controller
+	rcLister     cache.StoreToReplicationControllerLister
+
+	rsStore      cache.Store
+	rsController *framework.Controller
+	rsLister     cache.StoreToReplicaSetLister
+
+	dStore      cache.Store
+	dController *framework.Controller
+	dLister     cache.StoreToDeploymentLister
+
+	queue *workqueue.Type
+
+	getUpdater func() updater
+}
+
+// This is used to return (controller, scale) pairs from the controller finder
+// functions (see the next type).
```

Comment:
the "see next type" is probably not necessary and could break if someone accidentally inserts something in between, so I'd just remove it